client/src/components/CategoryShow.js

```diff
@@ -28,14 +48,57 @@ const CategoryShow = (props) => {
     getCategory();
   }, []);
 
+  const postPasta = async (newPastaData) => {
+    try {
+      const response = await fetch(`/api/v1/categories/${categoryId}/pastas`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(newPastaData)
+      })
+      console.log(response)
+      if (!response.ok) {
+          if(response.status === 422) {
+            const body = await response.json()
+            const newErrors = translateServerErrors(body.errors)
+            return setErrors(newErrors)
+          } else {
+            const errorMessage = `${response.status} (${response.statusText})`
+            const error = new Error(errorMessage)
+            throw(error)
+          }
+        } else {
+          const body = await response.json()
+          const updatedPastas = category.pastas.concat(body.pasta)
+          setErrors([])
+          setCategory({...category, pastas: updatedPastas})
+        }
+    } catch(error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
   let pastasList;
   if (!category.pastas[0]) {
     pastasList = `There are currently no pastas added to this category`;
   } else {
-    pastasList = category.pastas.map((pasta) => {
-      return <PastaTile key={pasta.id} {...pasta} />;
+    pastasList = category.pastas.map(pastaObject => {
+      return <PastaTile key={pastaObject.id} {...pastaObject} />;
     });
   }
+  const showForm = () => {
+    if (currentUser) {
```

Comment:
So we don't need/want to set up this entire page as an authenticated route - just need the user in order to show the form. For this, we can make use of `withRouter`
https://v5.reactrouter.com/web/api/withRouter

Wrap the component in a route from App.js
```js
// /components/App.js

<Route exact path="/categories/:id">
  <CategoryShow user={currentUser} />
</Route>
```
That way we can pass the component some props

Then in the show component, you'll make use of withRouter
```js
// /CategoryShow.js

import { withRoute } from "react-router"

...

export default withRouter(CategoryShow)
```
So that within the component you can use `props.match`, .etc as needed and access `currentUser`!